sdk/bls.go

```
@@ -0,0 +1,156 @@
+	sig := new(blst.P2Affine).Uncompress([]byte(signature))
+	return sig.Verify(true, pk, true, []byte(msg), []byte(FILECOIN_DST))
+}
+
+func (pub BLSPublicKey) HexString() string {
+	return strings.ToUpper(hex.EncodeToString([]byte(pub)))
+}
+
+func (signature BLSSignature) Verify(msg string, pub BLSPublicKey) bool {
+	pk := new(blst.P1Affine).Uncompress([]byte(pub))
+	sig := new(blst.P2Affine).Uncompress([]byte(signature))
+	return sig.Verify(true, pk, true, []byte(msg), []byte(FILECOIN_DST))
+}
+
+func (signature BLSSignature) HexString() string {
+	return strings.ToUpper(hex.EncodeToString([]byte(signature)))
+}
+
+type KeyPair struct {
+	PublicKey  BLSPublicKey
+	PrivateKey BLSPrivateKey
+}
+
+func GenerateKeyPair(seed io.Reader) *KeyPair {
+	sk := GeneratePrivateKey(seed)
+	pk := sk.Public()
+	return &KeyPair{
+		pk, sk,
+	}
+}
+
+func GenerateKeyPairFromIKM(ikm [32]byte) *KeyPair {
+	sk := GeneratePrivateKeyFromIKM(ikm)
+	pk := sk.Public()
+	return &KeyPair{
+		pk, sk,
+	}
+}
+
+func (p *KeyPair) Sign(msg string) BLSSignature {
+	return p.PrivateKey.Sign(msg)
+}
+
+func AggregateSignatures(signatures ...BLSSignature) (BLSSignature, error) {
+	if len(signatures) == 0 {
+		return ZeroBLSSignature, nil
+	}
+	temp := make([][]byte, len(signatures))
+	aggregator := new(blst.P2Aggregate)
+	for i, signature := range signatures {
+		temp[i] = []byte(signature)
+	}
+	if !aggregator.AggregateCompressed(temp, true) {
+		return ZeroBLSSignature, errors.New("unable to aggregate signature")
+	}
+
+	return BLSSignature(aggregator.ToAffine().Compress()), nil
+}
+
+func AggregatePublicKeys(keys ...BLSPublicKey) (BLSPublicKey, error) {
+	if len(keys) == 0 {
+		return ZeroBLSPublicKey, nil
+	}
+	temp := make([][]byte, len(keys))
+	aggregator := new(blst.P1Aggregate)
+	for i, key := range keys {
+		temp[i] = []byte(key)
+	}
+	if !aggregator.AggregateCompressed(temp, true) {
+		return ZeroBLSPublicKey, errors.New("unable to aggregate signature")
```

Are you sure that `AggregatePublicKeys` should return the error `unable to aggregate signature`?:D